### DISCUSSION TITLE
ID returned in wrong format

### DISCUSSION THREAD

**Discussion by gammelgaard52** [Art-of-WiFi/UniFi-Cloud-API-client#3] [Q&A] [answered]
I am not using this software in particular, but just extracting data through the API with PowerShell.
I have a problem where the data I get returned from https://api.ui.com/ea/hosts looks correct, however the id returned is in this format: abd19fc6-nnnn-nnnn-nnnn-e93d5b6886c6 but should be in: 900A6F00301100000000074A6BA90000000007A3387E0000000063EC9853

Any idea what to do?

This is my query:
$APIkey = "my-api-key-generated-in-site-manager-and-working"

$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
$headers.Add("Accept", "application/json")
$headers.Add("X-API-KEY", "$APIkey")

Invoke-WebRequest 'https://api.ui.com/ea/hosts' -Method 'GET' -Headers $headers -SessionVariable session

**Comment by malle-pietje** [comment]
@gammelgaard52 Ik think the pattern here is that these lowercase id's (UUIDs) are used for software-based controllers and the uppercase ones for UniFi OS-based consoles (CloudKeys, UDM's etc.) although there are a few exceptions to that rule.

It looks as if only the uppercase host id types are accepted as filter in the devices list. Can you confirm?

**Reply by malle-pietje** [reply]
After running more tests it looks as if some lowercase UUIDs can be used as filter for the `devices` endpoint.

**Reply by gammelgaard52** [reply]
Thank you for your quick reply. It led me to another round of code-reviewing

**Comment by malle-pietje** [comment]
Maybe something to ask on the UI Community with the "Site Manager" tag? Make sure to mention the UI-Team as well.

**Comment by gammelgaard52** [comment]
This is what I get when I run the invoke-webrequest. As you can see, the ID seems to me to be in the wrong format.
https://developer.ui.com/site-manager-api/get__ea_hosts/
{
  "data": [
    {
      "hardwareId": "78fa8073-nnnn-nnnn-nnnn-cffdfa2fnnnn",
      "id": "abd19fc6-nnnn-nnnn-nnnn-e93d5b68nnnn",
      "ipAddress": "removed",
      "isBlocked": false,
      "lastConnectionStateChange": "removed",
      "latestBackupTime": "removed",
      "owner": true,
      "registrationTime": "removed",
<OTHER REMOVED DATA>
      }
    }
  ],
  "httpStatusCode": 200,
  "traceId": "5fd7d84e44removedc91157d7"
}

**Reply by gammelgaard52** [reply]
I found the problem. Error 40. I should remove the curly brackets around the ID..

**Reply by gammelgaard52** [reply, marked as answer]
The ID is okay to not look like the example at UI API documentation.

**Reply by malle-pietje** [reply]
I agree this is quite confusing, hopefully someone from the Ubiquiti team is listening here? After all they are asking for feedback (but don't provide a channel?).
